pkg/reconciler/ingress/controller.go

```diff
@@ -168,3 +191,34 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 
 	return impl
 }
+
+func waitForCache(log *zap.SugaredLogger, caches *generator.Caches) {
+	timeout := time.After(cacheWarmUPTimeout)
+	tick := time.NewTicker(1 * time.Second)
+	defer tick.Stop()
+	for {
+		select {
+		case <-timeout:
+			log.Warnf("cache warm up timeout after %s", cacheWarmUPTimeout)
+			return
+		case <-tick.C:
+			if caches.HasSynced() {
+				return
+			}
+		}
+	}
+}
+
+func getReadyIngresses(knativeClient v1alpha12.NetworkingV1alpha1Interface) (map[string]struct{}, error) {
+	ingresses, err := knativeClient.Ingresses("").List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	ingressesToWarm := map[string]struct{}{}
+	for _, ingress := range ingresses.Items {
+		if ingress.Annotations[networking.IngressClassAnnotationKey] == config.KourierIngressClassName && ingress.GetStatus().GetCondition(v1alpha1.IngressConditionNetworkConfigured).IsTrue() {
```

Comment:
Not too important, but I think we could simplify this a bit extracting 2 helpers to `ingress.go`.
We could have something like: `IsManagedByKourier(ingress)` and `HasNetworkingConfigured(ingress)`

Reply:
Yes, let's add that in another PR.

Reply:
:+1: